test/sparse.cpp

```diff
@@ -271,3 +272,220 @@ TYPED_TEST(Sparse, EmptyDeepCopy) {
     EXPECT_TRUE(b.issparse());
     EXPECT_EQ(0, sparseGetNNZ(b));
 }
+
+TEST(Sparse, CPPSparseFromHostArrays)
+{
+    //! [ex_sparse_host_arrays]
+
+    float  vals[] = { 5, 8, 3, 6 };
+    int row_ptr[] = { 0, 0, 2, 3, 4 };
+    int col_idx[] = { 0, 1, 2, 1 };
+    const int M = 4, N = 4, nnz = 4;
+
+    array sparse = af::sparse(M, N, nnz, vals, row_ptr, col_idx, f32, AF_STORAGE_CSR);
+    // sparse
+    // Storage Format : AF_STORAGE_CSR
+    // [4 4 1 1]
+    // sparse: Values
+    // [4 1 1 1]
+    //     5.0000
+    //     8.0000
+    //     3.0000
+    //     6.0000
+
+    // sparse: RowIdx
+    // [5 1 1 1]
+    //          0
+    //          0
+    //          2
+    //          3
+    //          4
+
+    // sparse: ColIdx
+    // [4 1 1 1]
+    //          0
+    //          1
+    //          2
+    //          1
+
+    //! [ex_sparse_host_arrays]
+
+    array    vals_out = sparseGetValues(sparse);
+    array row_ptr_out = sparseGetRowIdx(sparse);
+    array col_idx_out = sparseGetColIdx(sparse);
+
+    array    vals_in = array(dim4(nnz,1), vals);
+    array row_ptr_in = array(dim4(M+1,1), row_ptr);
+    array col_idx_in = array(dim4(nnz,1), col_idx);
+
+    ASSERT_ARRAYS_EQ(vals_out, vals_in);
+    ASSERT_ARRAYS_EQ(row_ptr_out, row_ptr_in);
+    ASSERT_ARRAYS_EQ(col_idx_out, col_idx_in);
+}
+
+TEST(Sparse, CPPSparseFromAFArrays)
+{
+    //! [ex_sparse_af_arrays]
+
+    float v[] = { 5, 8, 3, 6 };
+    int   r[] = { 0, 0, 2, 3, 4 };
+    int   c[] = { 0, 1, 2, 1 };
+    const int M = 4, N = 4, nnz = 4;
+    array    vals = array(dim4(nnz), v);
+    array row_ptr = array(dim4(M+1), r);
+    array col_idx = array(dim4(nnz), c);
+
+    array sparse = af::sparse(M, N, vals, row_ptr, col_idx, AF_STORAGE_CSR);
+    // sparse
+    // Storage Format : AF_STORAGE_CSR
+    // [4 4 1 1]
+    // sparse: Values
+    // [4 1 1 1]
```

Comment:
Maybe display the sparse arrays as rows instead of columns. This takes up a lot of space.